### PR TITLE
Loadbalancer type of Service test for OCCM

### PR DESCRIPTION
**Pull request by lingxiankong**
Basically, the workflow is:
- A new PR related to OCCM code is submitted.
- Openlab launches a VM and runs this zuul job.
- The OCCM image is built inside the VM and uploaded to docker hub.
- The zuul job updates image for OCCM running in an existing k8s
  cluster on Catalyst Cloud.
- Zuul job executes a shell script as functional testing.

What are needed:
- An openrc file for Catalyst Cloud.
- The kubeconfig file for the existing k8s cluster.
- Put test-lb-service-catalyst-cloud.sh in the repo.

### CODE REVIEW

File: playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml [hunk 1]
```diff
@@ -1,124 +1,138 @@
...
         cmd: |
-          set -x
-          set -e
-          set -o pipefail
+          curl -LO https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl
+          chmod +x ./kubectl
+          mv ./kubectl /usr/local/bin/kubectl
+          mkdir ~/.kube
+          cat << EOF >> ~/.kube/config
+          {{ kubeconfig_content }}
```
Comment (bzhaoopenstack) [inline playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml]: This line needs to be updated with {{ kubeconfig.content }}, due to the zuul secret configuration limitation. It saids we have to define a dict format.

File: playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml [hunk 2]
```diff
@@ -1,124 +1,138 @@
...
-                echo "$test_case lb services launched failed!"
-                exit 1
-            fi
-          done
+    - name: Run acceptance tests for LoadBalancer Service
+      shell:
         executable: /bin/bash
-        chdir: '{{ k8s_os_provider_src_dir }}'
+        cmd: |
+          FLOATING_IP={{ floating_ip.stdout }} CLUSTERNAME=c886a618-c92a-4d4d-9f8f-bd2201f4dafd bash test-lb-service-catalyst-cloud.sh
```
Comment (bzhaoopenstack) [inline playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml]: test-lb-service-catalyst-cloud.sh This file I upload to the openlab-zuul-jobs/tools dir, For simplest, we can just download from there.

File: playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml (hunk 1 above)
Comment (lingxiankong) [reply playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml]: Done

File: playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml (hunk 2 above)
Comment (lingxiankong) [reply playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml]: I put the script into cloud-provider-openstack repo, it's more reasonable to maintain this script there. PR https://github.com/kubernetes/cloud-provider-openstack/pull/934

File: playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml [hunk 3]
```diff
@@ -1,124 +1,144 @@
...
         cmd: |
-          set -x
-          set -e
-          set -o pipefail
+          curl -LO https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl
+          chmod +x ./kubectl
+          mv ./kubectl /usr/local/bin/kubectl
+          mkdir ~/.kube
+          cat << EOF >> ~/.kube/config
+          {{ kubeconfig.content }}
```
Comment (bzhaoopenstack) [inline playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml]: Seems here, the kubeconfig secret is lost, please wait for the https://github.com/theopenlab/openlab-zuul-jobs/pull/793 merged.

Comment (bzhaoopenstack) [reply playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml]: And need to inject kubeconfig in the job definition.

Comment (bzhaoopenstack) [reply playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml]: https://github.com/theopenlab/openlab-zuul-jobs/blob/master/zuul.d/jobs.yaml#L822  . Once all of done, we have a try.

Comment (bzhaoopenstack) [reply playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml]: https://github.com/theopenlab/openlab-zuul-jobs/blob/master/zuul.d/jobs.yaml#L822  is merged.

**Pull request merged**